pkg/controller/buildrun/buildrun_controller_test.go

```diff
@@ -215,8 +215,54 @@ var _ = Describe("Reconcile BuildRun", func() {
 				Expect(err.Error()).To(ContainSubstring(fmt.Sprintf(" \"%s\" not found", strategyName)))
 			})
 
+			It("only generates the service account once if the task run cannot be created", func() {
+				// override the Build to use a cluster BuildStrategy
+				buildSample = ctl.DefaultBuild(buildName, strategyName, build.ClusterBuildStrategyKind)
+
+				// override buildrun to use a generated service account
+				buildRunName = "foobar-buildrun-sa-generate"
+				buildRunSample = ctl.BuildRunWithSAGenerate(buildRunName, buildName)
+				buildRunSample.Labels = make(map[string]string)
+				buildRunSample.Labels[build.LabelBuild] = buildName
+
+				// Override stub calls to include only build and buildrun
+				client.GetCalls(ctl.StubBuildRunGetWithoutSA(buildSample, buildRunSample))
+
+				// Call the reconciler
+				_, err := reconciler.Reconcile(request)
+
+				// Expect an error stating that the strategy does not exist
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(ContainSubstring(fmt.Sprintf(" \"%s\" not found", strategyName)))
+
+				// Expect one create call (service account)
+				Expect(client.CreateCallCount()).To(Equal(1))
+				_, obj, _ := client.CreateArgsForCall(0)
+				serviceAccount, castSuccessful := obj.(*corev1.ServiceAccount)
+				Expect(castSuccessful).To(BeTrue())
+
+				// Expect zero update calls
+				Expect(client.UpdateCallCount()).To(Equal(0))
+
+				// Change the Get stub to also return the service account
+				client.GetCalls(ctl.StubBuildRunGetWithSA(buildSample, buildRunSample, serviceAccount))
+
+				// Call the reconciler again
+				_, err = reconciler.Reconcile(request)
+
+				// Expect an error stating that the strategy does not exist
```

Comment:
Why there are two places to check `Expect an error stating that the strategy does not exist` in this new added test?

I would like to know what result you want to expect?

Reply:
This unit test is testing the behavior described in the issue #203. The first reconcilation fails with the expected failure (the non-existing build strategy). The second reconcilation before the fix failed with the error stating that the service account cannot be created. This was fixed and the "real" error is still happening. This is verified here.

Reply:
OK, thanks for the intro